`RememberMeMigration` creates `rememberme_token` table without configured charset & collation

### Affected version(s)

5.3+

### Description

Our `RememberMeMigration` currently creates the `rememberme_token` table without a collation:

https://github.com/contao/contao/blob/d2247a16c7f8bf55d340118f12f54a03497e0e0d/core-bundle/src/Migration/Version503/RememberMeMigration.php#L38-L44

Meaning that whatever default collation is configured on the current MySQL server will be used instead. This can create issues - for example if you want to restore a database backup on another system where that specific default collation of the source system is not available:

```
$ bin/console c:b:r


 [ERROR] An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1273 Unknown collation:                 
         'utf8mb4_uca1400_ai_ci'  
```

Our `CREATE TABLE` statement should instead include the configured default charset and collation for the current database connection, to keep the table consistent with all the others.

In a fresh Contao 5.3+ installation (i.e. when not migrating from Contao 4.13) the `rememberme_token` table will be created correctly with the configured charset and collation.

--- a/core-bundle/config/migrations.yaml
+++ b/core-bundle/config/migrations.yaml
@@ -80,4 +80,3 @@ services:
         class: Contao\CoreBundle\Migration\Version503\RememberMeMigration
         arguments:
             - '@database_connection'
-            - !tagged_iterator security.remember_me_handler

--- a/core-bundle/src/Migration/Version503/RememberMeMigration.php
+++ b/core-bundle/src/Migration/Version503/RememberMeMigration.php
@@ -15,6 +15,8 @@ namespace Contao\CoreBundle\Migration\Version503;
 use Contao\CoreBundle\Migration\AbstractMigration;
 use Contao\CoreBundle\Migration\MigrationResult;
 use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Types;
 
 /**
  * @internal
@@ -34,18 +36,30 @@ class RememberMeMigration extends AbstractMigration
 
     public function run(): MigrationResult
     {
-        $this->connection->executeStatement(<<<'SQL'
-            CREATE TABLE `rememberme_token` (
-                `series`   varchar(88)  UNIQUE PRIMARY KEY NOT NULL,
-                `value`    varchar(88)  NOT NULL,
-                `lastUsed` datetime     NOT NULL COMMENT '(DC2Type:datetime_immutable)',
-                `class`    varchar(100) NOT NULL,
-                `username` varchar(200) NOT NULL
-            );
-            SQL,
-        );
+        $table = new Table('rememberme_token');
+        $table->addColumn('series', Types::STRING, ['length' => 88]);
+        $table->addColumn('value', Types::STRING, ['length' => 88]);
+        $table->addColumn('lastUsed', Types::DATETIME_IMMUTABLE);
+        $table->addColumn('class', Types::STRING, ['length' => 100]);
+        $table->addColumn('username', Types::STRING, ['length' => 200]);
+        $table->setPrimaryKey(['series']);
+
+        $params = $this->connection->getParams()['defaultTableOptions'] ?? [];
+
+        if (isset($params['charset'])) {
+            $table->addOption('charset', $params['charset']);
+        }
+
+        if (isset($params['engine'])) {
+            $table->addOption('engine', $params['engine']);
+        }
+
+        if (isset($params['collate'])) {
+            $table->addOption('collate', $params['collate']);
+        }
 
         $schemaManager = $this->connection->createSchemaManager();
+        $schemaManager->createTable($table);
 
         // If tl_remember_me.userIdentifier exists, the database is from Contao 5 and the
         // existing tokens can be migrated. Otherwise, it is a Contao 4 database and the

